x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -247,36 +247,47 @@ const getProductTypes = (
 export const cli = () => {
   run(
     async (context) => {
+      const environment = process.env.ENVIRONMENT ? `${process.env.ENVIRONMENT}` : 'qa';
+
+      const PROXY_URL = process.env.PROXY_URL ? process.env.PROXY_URL : undefined;
+      const PROXY_SECRET = process.env.PROXY_SECRET || undefined;
+      const PROXY_CLIENT_ID = process.env.PROXY_CLIENT_ID || undefined;
+
       // Checking if API key is either provided via env variable or in ~/.elastic.cloud.json
       // This works for either local executions or fallback in case proxy service is unavailable.
-      if (!process.env.CLOUD_QA_API_KEY && !getApiKeyFromElasticCloudJsonFile()) {
-        log.error('The API key for the environment needs to be provided with the env var API_KEY.');
+      if (!process.env.CLOUD_API_KEY && !getApiKeyFromElasticCloudJsonFile(environment)) {
+        log.error(
+          'The API key for the environment needs to be provided with the env var CLOUD_API_KEY.'
+        );
         log.error(
           'If running locally, ~/.elastic/cloud.json is attempted to be read which contains the API key.'
         );
         // eslint-disable-next-line no-process-exit
         return process.exit(1);
       }
-
-      const PROXY_URL = process.env.PROXY_URL ? process.env.PROXY_URL : undefined;
-      const PROXY_SECRET = process.env.PROXY_SECRET ? process.env.PROXY_SECRET : undefined;
-      const PROXY_CLIENT_ID = process.env.PROXY_CLIENT_ID ? process.env.PROXY_CLIENT_ID : undefined;
-
-      const API_KEY = process.env.CLOUD_QA_API_KEY
-        ? process.env.CLOUD_QA_API_KEY
-        : getApiKeyFromElasticCloudJsonFile();
+      const API_KEY = process.env.CLOUD_API_KEY
+        ? process.env.CLOUD_API_KEY
+        : getApiKeyFromElasticCloudJsonFile(environment);
 
       log.info(`PROXY_URL is defined : ${PROXY_URL !== undefined}`);
       log.info(`PROXY_CLIENT_ID is defined : ${PROXY_CLIENT_ID !== undefined}`);
       log.info(`PROXY_SECRET is defined : ${PROXY_SECRET !== undefined}`);
       log.info(`API_KEY is defined : ${API_KEY !== undefined}`);
 
       let cloudHandler: ProjectHandler;
-      if (PROXY_URL && PROXY_CLIENT_ID && PROXY_SECRET && (await proxyHealthcheck(PROXY_URL))) {
+      const proxyServiceUse =
+        PROXY_URL &&
+        PROXY_CLIENT_ID &&
+        PROXY_SECRET &&
+        environment.toUpperCase() !== 'prod'.toUpperCase() &&
```

Comment:
Why casing normalization is only used here? Don't we need to do the same in `getApiKeyFromElasticCloudJsonFile()`? I'd recommend normalize `environment`'s value at the beginning on the line `250`. It will help to match case insensitively. And it should match with my comment above regarding known `environment` values.



`environment.toUpperCase() !== 'prod'.toUpperCase()` can be statically transformed to `environment.toUpperCase() !== 'PROD'`. Though why casing is only used here but

Reply:
Agreed!